cmd/kubeadm/app/phases/controlplane/users_linux.go

```diff
@@ -0,0 +1,260 @@
+// +build linux
+
+package controlplane
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"strconv"
+	"strings"
+
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/utils/exec"
+)
+
+var ()
+
+var UpdateFileOwnershipFn = func(filepath string, uid int64, gid int64, permissions uint32) error {
+	if err := os.Chown(filepath, int(uid), int(gid)); err != nil {
+		return err
+	}
+	if err := os.Chmod(filepath, fs.FileMode(permissions)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func UpdateFileOwnership(filepath string, uid int64, gid int64, permissions uint32) error {
+	return UpdateFileOwnershipFn(filepath, uid, gid, permissions)
+}
+
+var UpdateDirecotryOwnershipFn = func(directorypath string, uid int64, gid int64, permissions string) error {
+	exec := exec.New()
+	cmd := exec.Command("chown", "-R", fmt.Sprintf("%d:%d", uid, gid), directorypath)
```

Comment:
Add a comment here explaining why we chose to run chown using shell and not using os.Chown function, it is because os.Chown does not support recursively updating the permissions of all files in the directory. So we would have to walk all the directories and subdirectories, it is more straight forward to just use `chown -R` instead.

Reply:
the chances of the CLI too `chown` missing on a host are minimal, but we should avoid shelling out to more tools and use the syscall that the go stdlib exposes.

the recursive directory walk should be a fairly low number of extra LOC.